pkg/agent/route/route_linux.go

```diff
@@ -562,43 +600,108 @@ func (c *Client) writeEKSNATRules(iptablesData *bytes.Buffer) {
 	}...)
 }
 
+func (c *Client) getIPProtocol() iptables.Protocol {
+	switch {
+	case c.networkConfig.IPv4Enabled && c.networkConfig.IPv6Enabled:
+		return iptables.ProtocolDual
+	case c.networkConfig.IPv6Enabled:
+		return iptables.ProtocolIPv6
+	default:
+		return iptables.ProtocolIPv4
+	}
+}
+
+// Create the Antrea managed chains and link them to built-in chains.
+// We cannot use iptables-restore for these jump rules because there
+// are non Antrea managed rules in built-in chains.
+type jumpRule struct {
+	table    string
+	srcChain string
+	dstChain string
+	comment  string
+	insert   bool
+}
+
+func (c *Client) ensureAntreaJumpRulePrecedence(protocol iptables.Protocol, jumpRule jumpRule) error {
```

Comment:
The function name is misleading, it doesn't really ensure precedence but removes the rule under some conditions.

Reply:
Renamed it to `removeUnexpectedAntreaJumpRule`